components/nimbus/src/lib.rs

```diff
@@ -387,9 +396,13 @@ impl Experiment {
 #[serde(rename_all = "camelCase")]
 pub struct FeatureConfig {
     pub feature_id: String,
-    pub enabled: bool,
+    // Backward Compatibility note: Originally, this used to be
+    // required, so converting it into an option should not break
+    // backward compatibility since old experiments are guaranteed
+    // to have this field
+    pub enabled: Option<bool>,
```

Comment:
`enabled` became nullable and not completely removed... we can opt to remove it since we aren't using it, but I wasn't sure how close to the `nimbus-shared` schema we want to stick


As for removing the fields in general, I'm getting hit by a test `test_migrate_db_v1_to_db_v2_round_tripping` that expects a persisted `v1` `json` to stay the same after getting deserialized and serialized... Is that a type of backward compatibility we want to keep? Or is the ability to deserialize the only thing we care about?

Because if we remove the fields, when we deserialize and serialize the `v1` experiments, they would no longer be the **exact** same `json`. I'm happy to modify that test, to only verify that `v1` experiments can be deserialized, and I can't see any bad implications from that, but was being extra careful.

Either way, I agree that the scope here is big, so I'll split that into its own PR